node-js-basics-101/node_modules/asynquence/tests.js

```diff
@@ -0,0 +1,1528 @@
+				sq1, sq2, seed = 10
+			;
+
+			function seqMessages(msg1,msg2,msg3) {
+				return ASQ(function(done){
+					setTimeout(function(){
+						done(msg1,msg2,msg3);
+					},25);
+				});
+			}
+
+			function promiseMessages(msg1,msg2,msg3) {
+				return new Promise(function(resolve){
+					setTimeout(function(){
+						resolve(ASQ.messages(msg1,msg2,msg3));
+					},25);
+				});
+			}
+
+			sq1 = ASQ()
+			.seq(seqMessages)
+			.promise(promiseMessages)
+			.val(function(s1,s2,s3){
+				// if any messages received, use them
+				seed += ((s1 + s2 + s3) || 0);
+			})
+			.then(asyncDelayFn(100))
+			.then(function(done){
+				done(++seed);
+			})
+			.gate(
+				function(done){
+					done(++seed);
+				},
+				function(done){
+					done(++seed);
+				}
+			);
+
+			// duplicate a template of the sequence
+			sq2 = sq1.duplicate();
+
+			sq1.val(function(msg1,msg2){
+				if (!(
+					msg1 === 12 &&
+					msg2 === 13
+				)) {
+					clearTimeout(timeout);
+					var args = ARRAY_SLICE.call(arguments);
+					args.unshift(testDone,label);
+					FAIL.apply(FAIL,args);
+				}
+			})
+			.then(function(){
+				seed = 20;
+
+				// unpause the duplicated sequence
+				sq2 = ASQ.unpause(sq2);
+				// inject some messages into the unpausing sequence
+				//   hint: not a great idea, but supported
+				sq2.unpause(1,2);
+				sq2.unpause(3);
+
+				// later, check to see if the sequence
+				// was indeed restarted and ran properly
+				setTimeout(function(){
+					if (seed !== 29) {
+						clearTimeout(timeout);
+						var args = [testDone,label,"seed: " + seed];
+						FAIL.apply(FAIL,args);
+						return;
+					}
+
+					sq2.val(function(msg1,msg2){
+						if (!(
+							msg1 === 28 &&
+							msg2 === 29
+						)) {
+							clearTimeout(timeout);
+							var args = ARRAY_SLICE.call(arguments);
+							args.unshift(testDone,label);
+							FAIL.apply(FAIL,args);
+						}
+					})
+					.then(function(){
+						clearTimeout(timeout);
+						PASS(testDone,label);
+					})
+					.or(function(){
+						clearTimeout(timeout);
+						var args = ARRAY_SLICE.call(arguments);
+						args.unshift(testDone,label);
+						FAIL.apply(FAIL,args);
+					});
+				},500);
+			})
+			.or(function(){
+				clearTimeout(timeout);
+				var args = ARRAY_SLICE.call(arguments);
+				args.unshift(testDone,label);
+				FAIL.apply(FAIL,args);
+			});
+
+			timeout = setTimeout(function(){
+				FAIL(testDone,label + " (from timeout)");
+			},2000);
+		});
+		tests.push(function(testDone){
+			var label = "Core Test #24", timeout, ASQ2, ASQ3;
+
+			ASQ2 = ASQ.clone();
+
+			ASQ2._hello_ = "world";
+			ASQ2.extend("foobar",function(){ return function(){}; });
+
+			ASQ3 = ASQ2.clone();
+
+			try {
+				ASQ().foobar();
+				FAIL(testDone,label,"ASQ().foobar()");
+				return;
+			} catch (err) {}
+
+			try {
+				ASQ2().foobar();
+			}
+			catch (err) {
+				FAIL(testDone,label,"ASQ2().foobar()",err,ASQ2(),ASQ2().foobar);
+				return;
+			}
+
+			try {
+				ASQ3._hello_.length;
```

Comment:
__[jshint]__
* Expected an assignment or function call and instead saw an expression.

<!-- Comment by [SideCI](https://sideci.com) -->